config/config.go

```diff
@@ -260,14 +260,18 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	// Do global overrides and validate unique names.
 	jobNames := map[string]struct{}{}
 	for _, scfg := range c.ScrapeConfigs {
+		if scfg.ScrapeTimeout > scfg.ScrapeInterval {
```

Comment:
I'm fine with having the options. I'm not so fine with setting a timeout also requiring setting the same interval as is specified in the global config, as that increases the chances that over time users will think it's perfectly okay to have different scrape intervals inside the same Prometheus.

I propose we keep this logic as-is.